cmd/cluster-watchall/main.go

```diff
@@ -61,34 +60,13 @@ import (
    for each one of those, all objects in that cluster.
 */
 
-type ClientOpts struct {
-	loadingRules *clientcmd.ClientConfigLoadingRules
-	overrides    clientcmd.ConfigOverrides
-}
-
-func NewClientOpts() *ClientOpts {
-	return &ClientOpts{
-		loadingRules: clientcmd.NewDefaultClientConfigLoadingRules(),
-		overrides:    clientcmd.ConfigOverrides{},
-	}
-}
-
-func (opts *ClientOpts) AddFlags(flags *pflag.FlagSet) {
-	flags.StringVar(&opts.loadingRules.ExplicitPath, "kubeconfig", opts.loadingRules.ExplicitPath, "Path to the kubeconfig file to use for CLI requests")
-	flags.StringVar(&opts.overrides.CurrentContext, "context", opts.overrides.CurrentContext, "The name of the kubeconfig context to use")
-}
-
-func (opts *ClientOpts) ToRESTConfig() (*rest.Config, error) {
-	clientConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(opts.loadingRules, &opts.overrides)
-	return clientConfig.ClientConfig()
-}
-
 func main() {
 	serverBindAddress := ":10203"
 	fs := pflag.NewFlagSet("watchall", pflag.ExitOnError)
 	klog.InitFlags(flag.CommandLine)
 	fs.AddGoFlagSet(flag.CommandLine)
-	cliOpts := NewClientOpts()
+	cliOpts := genericclioptions.NewConfigFlags(false)
```

Comment:
Maybe I'm missing something here , genericclioptions.NewConfigFlags() returns ConfigFlags. I guess that if it passes compilation it implements all the functions you had (and still using) for ClientOpts ...
nit: maybe change the name to cliFlags ?

Reply:
Not coincidentally, the old `NewClientOpts` func produced something that behaves much like what `genericclioptions.NewConfigFlags` returns.

I changed the variable name as suggested.